src/gridfs/upload.ts

```diff
@@ -143,7 +149,7 @@ export class GridFSBucketWriteStream extends Writable {
   abort(callback: Callback<void>): void;
   abort(callback?: Callback<void>): Promise<void> | void {
     const Promise = PromiseProvider.get();
-    let error: MongoDriverError;
+    let error: MongoGridFSStreamError;
```

Comment:
Can we refactor this to use `maybePromise`?

Reply:
In hindsight, I think this whole thing is out of scope for this ticket. I `MongoStreamClosedError`, covered in [NODE-3405](https://github.com/mongodb/node-mongodb-native/pull/2913), better covers the errors thrown here. I'll leave a TODO comment for that PR and revert this change.